src/Package.appxmanifest

```diff
@@ -8,6 +8,22 @@
         <Interface Name="Microsoft.Windows.Widgets.Providers.IWidgetProvider2" InterfaceId="38C3A963-DD93-479D-9276-04BF84EE1816" />
       </ProxyStub>
     </Extension>
+    <!-- QuietBackgroundProcesses unelevated server which initiates & obtains a reference to a quiet session -->
+    <Extension Category="windows.activatableClass.outOfProcessServer">
+      <OutOfProcessServer ServerName="DevHome.QuietBackgroundProcesses.Server" uap5:IdentityType="activateAsPackage" uap5:RunFullTrust="true">
+        <Path>DevHome.QuietBackgroundProcesses.Server.exe</Path>
+        <Instancing>singleInstance</Instancing>
+        <ActivatableClass ActivatableClassId="DevHome.QuietBackgroundProcesses.QuietBackgroundProcessesSessionManager" />
+      </OutOfProcessServer>
+    </Extension>
+    <!-- QuietBackgroundProcesses elevated server owns the quiet session -->
+    <Extension Category="windows.activatableClass.outOfProcessServer">
+      <OutOfProcessServer ServerName="DevHome.QuietBackgroundProcesses.ElevatedServer" uap5:IdentityType="activateAsPackage" uap5:RunFullTrust="true">

```

Comment:
>ElevatedServer [](http://example.com/codeflow?start=71&length=14)

2 things to consider from a security perspective:
1. Should this have a security descriptor to scope access?
2. Because the class configuration doesn't support explicitly baking in "this class is elevated", a non-elevated process (running under this package's identity and at medium IL) could spoof this class. Does that present a concerning threat? I don't think it does (e.g. you're not receiving secrets from high-trust processes or anything like that).

Reply:
Great point. I thought about it, and I think you're right that's it's a hole (that nothing of sensitivity will go through yet, but) that should be closed.

So how do I do it for a WinRT class?

For COM I'd do this, but that's not supported for WinRT classes in the appxmanifest.
    LaunchAndActivationPermission="O:BAG:BAD:(A;;CCSW;;;BU)">

Do I have to use registry.dat?

Reply:
Oh. No, we don't actually expose manifest syntax that would allow this; the underlying code handles permissions, you just can't get deployment to emit them.

Anyone who can put your package on their dependency graph can activate your server as long as they can pass this default security descriptor we make up by default:

O:PSG:BAD:(A;;0xB;;;SY)(A;;0xB;;;BA)(A;;0xB;;;BU)(A;;0xB;;;<your package's AppContainer SID>)S:(ML;;NX;;;LW)

Reply:
> 1. Should this have a security descriptor to scope access?
> Oh. No, we don't actually expose manifest syntax that would allow [specifying an SD]; 

I don't know how to reconcile these 2 statements.  I would do it if it's possible - but seems like it's not.

 > Anyone who can put your package on their dependency graph can activate your server as long as they can pass this default security descriptor

Right, but we don't exactly care who activates it as long as it requires admin to be activated.  Ideally the manifest syntax would let me specify an SD that requires that "whoever calls RoRegisterActivationFactories is an elevated token" - because then everyone would _know_ that the server is elevated (i.e. if an [elevated] someone spoofed it, then who cares because they're already godlike.)

I'm not sure how to move forward to fix the original concern.

Reply:
> I don't know how to reconcile these 2 statements.  

You don't. I made a suggestion, and subsequently realized it was impossible with the current OS code. 

> Right, but we don't exactly care who activates it as long as it requires admin to be activated.  Ideally the manifest syntax would let me specify an SD that requires that "whoever calls RoRegisterActivationFactories is an elevated token" - because then everyone would know that the server is elevated (i.e. if an [elevated] someone spoofed it, then who cares because they're already godlike.)

Yep, that would be ideal, but can't currently be done.

Bottom line:
Clients need to consider the possibility that the server is spoofed; I think the API design means spoofs aren't particularly threatening to the client, so no action may be needed.
The server needs to consider the possibility that its client is lower-privilege than it should be, since it can't customize its SD. If that represents a genuine threat, then the server can mitigate it by checking its caller's identity at runtime (see [https://www.osgwiki.com/wiki/COM_Caller_Information](https://www.osgwiki.com/wiki/COM_Caller_Information))

If neither of those represent interesting threats, then nothing may be required here.